src/mc_robots/CMakeLists.txt

```diff
@@ -24,8 +24,14 @@ macro(generate_aliases PATH VAR NAME FILE_IN FILE_OUT)
     list(APPEND ${VAR} "${NAME}/${URDF_NAME}: [${NAME}, ${PATH}, ${URDF_NAME}]")
   endforeach()
   string(REPLACE ";" "\n" ${VAR} "${${VAR}}")
-  configure_file("${FILE_IN}" "${CMAKE_CURRENT_BINARY_DIR}/${FILE_OUT}")
-  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${FILE_OUT}" DESTINATION "${MC_ROBOTS_INSTALL_PREFIX}/aliases/")
+
+  if(CMAKE_CONFIGURATION_TYPES)
+    set(PATH_OUT "${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE}/aliases/${FILE_OUT}")
```

Comment:
I think you want to use `$<CONFIGURATION>` here, `CMAKE_BUILD_TYPE` is likely empty if `CMAKE_CONFIGURATION_TYPES` is defined. Although I don't think there's any difference between the generated aliases between build types so I don't we need to bother with this.

Reply:
`$<CONFIGURATION>` is not usable in that context (I naively tried), as this is not a variable. This works fine for the CI at least, but yeah, probably not optimal.

> Although I don't think there's any difference between the generated aliases between build types so I don't we need to bother with this.

The issue here is that the path for aliases is defined as that of the configuration's `RobotModulePath + /alias`, and on windows builds the `RobotModulePath` points to the corresponding build type (`build/src/mc_robots/RelWithDebInfo`). So in order for mc_rtc to find the aliases files, they need to be installed in that folder as well (`build/src/mc_robots/RelWithDebInfo/aliases`), hence the current patch.

Reply:
Ok, gotcha!

In that case, you can use `configure_file` as before and `file(GENERATE)` (see https://cmake.org/cmake/help/latest/command/file.html#generate) to put them in the right folder. Fortunately, this later commands understand generator expressions

> This works fine for the CI at least, but yeah, probably not optimal.

This is because the CI script always pass `-DCMAKE_BUILD_TYPE` but I think it will break in a normal multi-configuration setup.

Reply:
Thanks, I wasn't familiar with `file(GENERATE)`. Should be good now.